File: Python/Program/Tools.py
from Equipment import Equipment

class Tools(Equipment):
    def __init__(self, name: str, purpose: str, cost: int, ingredient: str, improvement: str):
        super().__init__(name, purpose)
        self._cost = cost
        self._ingredient = ingredient
        self._improvement = improvement

    def get_cost(self):
        return self._cost

    def set_cost(self, cost: int):
        self._cost = cost

    def get_ingredient(self):
        return self._ingredient

    def set_ingredient(self, ingredient: str):
        self._ingredient = ingredient

    def get_improvement(self):
        return self._improvement

    def set_improvement(self, improvement: str):
        self._improvement = improvement

File: C++/Program/Tools.cpp
#pragma once
#include <iostream>
#include <string>
#include "Equipment.cpp"

using namespace std;

class Tools : public Equipment {
private:
    int cost;
    string ingredient;
    string improvement;

public:
    // Constructor
    Tools(){}

    Tools(string name, string purpose, int cost, string ingredient, string improvement)
        : Equipment(name, purpose) {
        this->cost = cost;
        this->ingredient = ingredient;
        this->improvement = improvement;
    }

    // Getter and Setter for cost
    int getCost() {
        return cost;
    }

    void setCost(int cost) {
        this->cost = cost;
    }

    // Getter and Setter for ingredient
    string getIngredient() {
        return ingredient;
    }

    void setIngredient(string ingredient) {
        this->ingredient = ingredient;
    }

    // Getter and Setter for improvement
    string getImprovement() {
        return improvement;
    }

    void setImprovement(string improvement) {
        this->improvement = improvement;
    }
};